### PR TITLE
Spells/DK: Fix Heartbreaker talent

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Fix Heartbreaker talent

**Tests performed:**

Tested in-game

### CODE REVIEW

File: src/server/scripts/Spells/spell_dk.cpp [hunk 1]
```diff
@@ -732,6 +734,32 @@ class spell_dk_glyph_of_scourge_strike_script : public SpellScript
...
+    }
+
+    bool Load() override
+    {
+        return GetCaster()->HasAura(SPELL_DK_HEARTBREAKER_TALENT);
+    }
+
+    void HandleEnergize(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetCaster(), SPELL_DK_HEARTBREAKER_ENERGIZE, CastSpellExtraArgs()
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_dk.cpp:752]: `Heart Strike generates ${$210738s1/10} additional Runic Power per target hit.` should only be 2, not 20 per hit target

File: src/server/scripts/Spells/spell_dk.cpp [hunk 2]
```diff
@@ -732,6 +734,32 @@ class spell_dk_glyph_of_scourge_strike_script : public SpellScript
...
+            .SetTriggeringSpell(GetSpell())
+            .SetTriggerFlags(TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR));
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_dk_heartbreaker::HandleEnergize, EFFECT_0, SPELL_EFFECT_DUMMY);
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_dk.cpp:759]: hook it on EFFECT_1 (SPELL_EFFECT_SCHOOL_DAMAGE), thats the Effect with ChainTargets

File: sql/updates/world/master/2024_99_99_99_dk_heartbreaker.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM `spell_script_names` WHERE `spell_id`=206930 AND `ScriptName`='spell_dk_heartbreaker';
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES (206930, 'spell_dk_heartbreaker');
```
Comment (mdX7) [inline sql/updates/world/master/2024_99_99_99_dk_heartbreaker.sql:2]: newline after `VALUES`

File: src/server/scripts/Spells/spell_dk.cpp (hunk 1 above)
Comment (aquadeus) [reply src/server/scripts/Spells/spell_dk.cpp:752]: is hitting 2

File: src/server/scripts/Spells/spell_dk.cpp (hunk 2 above)
Comment (mdX7) [reply src/server/scripts/Spells/spell_dk.cpp:759]: turns out my SpellWork was bugged, EFFECT_0 is also using ChainTargets

Comment (aquadeus) [reply src/server/scripts/Spells/spell_dk.cpp:759]: bad boi

**Pull request merged**